app/validators/action_kit_fields.rb

```diff
@@ -0,0 +1,94 @@
+# The <tt>ActionKitFields</tt> class is a Rails <tt>ActiveModel::Validator<tt> class
+# for validating a model's +name+ field against ActionKits +action+ and +user+
+# tables.
+#
+# Valid field names must be whitelisted ( see <tt>ACTIONKIT_FIELDS_WHITELIST</tt> ),
+# or they must be prefixed with +user_+ or +action_+. All field names must not
+# have spaces or dashes.
+#
+# See also https://act.sumofus.org/docs/manual/api/rest/actionprocessing.html for detailed documentation
+# on ActionKit's action processing.
+#
+# ==== Valid Field Names
+#
+#   address1
+#   state
+#   action_bar
+#   action_foo_bar
+#   user_foo
+#
+# ==== Invalid Field Names
+#
+#   address_action
+#   foo-bar
+#   foobar
+#   action_foo bar
+#
+
+class ActionKitFields < ActiveModel::Validator
+  VALID_CHARS_RE = /^[0-9a-z_]+$/
+
+  VALID_PREFIX_RE = /^(action|user)\_[0-9a-z_]+/
+
+  ACTIONKIT_FIELDS_WHITELIST = %w{
+    address1
+    address2
+    city
+    country
+    email
+    first_name
+    home_phone
+    last_name
+    middle_name
+    mobile_phone
+    name
+    password
+    phone
+    plus4
+    postal
+    prefix
+    region
+    state
+    subscription_status
+    suffix
+    zip
+  }
```

Comment:
Nope, don't want to discuss product design stuff here in depth, as that should happen substantively in Pivotal (or through conversation) in my view, not through code review. Tara, let's touch base on all of this tomorrow when we chat verbally.